### PR TITLE
Fix some test issues in Spark UT and keep RapidsTestSettings update-to-date

### PR DESCRIPTION
**Pull request by thirtiseven**
Contributes to #10771 #10773 #10775

This PR:
- Updated issue links in Spark UT Test settings to specific ones.
- Ported two cases in RapidsCastSuite because of a spark issue https://issues.apache.org/jira/browse/SPARK-40851, the 330 test cases failed under new version of jdk8, so updated to a newer version of this test to make it pass.
- Removed UTC settings in TestsBaseTrait to fix three issues in RapidsJsonSuite, Spark use "America/Los_Angeles" as default timezone so we should not set additional utc config. [spark code](https://github.com/apache/spark/blob/334816af5f865d0e4c7a8e9c02b30236052d214e/core/src/test/scala/org/apache/spark/SparkFunSuite.scala#L85).

### CODE REVIEW

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build

File: tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsTestsTrait.scala [hunk 1]
```diff
@@ -102,8 +101,6 @@ trait RapidsTestsTrait extends RapidsTestsCommonTrait {
         .config("spark.sql.queryExecutionListeners",
           "org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback")
         .config("spark.sql.warehouse.dir", warehouse)
```
Comment (binmahone) [inline tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsTestsTrait.scala:103]: should also take care of RapidsSQLTestsBaseTrait

Comment (binmahone) [reply tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsTestsTrait.scala:103]: my mistake

**Comment by thirtiseven** [comment]
Premerge failed because the tests still think timezone is UTC, maybe caused by some settings in CI. Hardcoded to LA and trying again.

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build

**Comment by GaryShen2008** [comment]
After removing UTC config, I found the new added UT cases passed by falling back to CPU. So, I think we may not need to copy them out.

**Comment by thirtiseven** [comment]
Hi @GaryShen2008 , what’s your java version? The new added cases can pass under older Java version but fail on newer ones.

File: tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonExpressionsSuite.scala [hunk 2]
```diff
@@ -19,8 +19,16 @@
...
 import org.apache.spark.sql.catalyst.expressions.JsonExpressionsSuite
 import org.apache.spark.sql.rapids.utils.{RapidsJsonConfTrait, RapidsTestsTrait}
 
 class RapidsJsonExpressionsSuite
-  extends JsonExpressionsSuite with RapidsTestsTrait with RapidsJsonConfTrait {}
+    extends JsonExpressionsSuite with RapidsTestsTrait with RapidsJsonConfTrait {
+  override def beforeAll(): Unit = {
+      super.beforeAll()
+      // Set timezone to UTC to avoid fallback, so that tests run on GPU to detect bugs
+      TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
```
Comment (binmahone) [inline tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonExpressionsSuite.scala:32]: set it back in AfterAll()?

File: tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonFunctionsSuite.scala [hunk 3]
```diff
@@ -19,8 +19,16 @@
...
 import org.apache.spark.sql.JsonFunctionsSuite
 import org.apache.spark.sql.rapids.utils.{RapidsJsonConfTrait, RapidsSQLTestsTrait}
 
 class RapidsJsonFunctionsSuite
-  extends JsonFunctionsSuite with RapidsSQLTestsTrait with RapidsJsonConfTrait {}
+    extends JsonFunctionsSuite with RapidsSQLTestsTrait with RapidsJsonConfTrait {
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    // Set timezone to UTC to avoid fallback, so that tests run on GPU to detect bugs
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
```
Comment (binmahone) [inline tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonFunctionsSuite.scala:32]: set it back in AfterAll()?

File: tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonExpressionsSuite.scala (hunk 2 above)
Comment (thirtiseven) [reply tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonExpressionsSuite.scala:33]: done

File: tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonFunctionsSuite.scala (hunk 3 above)
Comment (thirtiseven) [reply tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsJsonFunctionsSuite.scala:35]: done

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build

**Comment by binmahone** [comment]
LGTM

**Review by binmahone — APPROVED** [review]

**Pull request merged**